app/services/patient_deduplication/strategies.rb

```diff
@@ -14,16 +16,27 @@ def identifier_and_full_name_match
       end
 
       # Exact match of just identifiers, excluding exact name matches
-      def identifier_excluding_full_name_match(limit: nil)
+      # where at least one duplicate patient belongs to given facilities.
+      def identifier_excluding_full_name_match(limit: nil, facilities: [])
         matches = PatientBusinessIdentifier
           .select("identifier, array_agg(patient_id) as patient_ids")
           .joins(:patient)
-          .where.not(identifier: "")
-          .where(identifier_type: simple_bp_passport)
           .group("identifier")
           .having("COUNT(distinct patient_id) > 1")
           .having("COUNT(distinct lower(full_name)) > 1")
 
+        if facilities.present?
+          identifiers_for_facilities =
+            PatientBusinessIdentifier
+              .joins(:patient)
+              .where(patients: {assigned_facility: facilities})
+              .where.not(identifier: "")
+              .where(identifier_type: simple_bp_passport)
+              .pluck("identifier")
+
+          matches = matches.where(identifier: identifiers_for_facilities)
```

Comment:
We should leave a note somewhere about the implementation being tailored for performance. Else, this is bound to get refactored especially since there aren't any tests either.